File: Merge_Intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {

        int n= intervals.size();
        vector<vector<int>> ans;

        sort(intervals.begin(),intervals.end());

        for(int i = 0 ; i<n;i++){
            if(ans.empty()){
              ans.push_back(intervals[i]);
            }else{
                
                vector<int> &v= ans.back();
                int x=v[1];

                if(intervals[i][0]<=x){
                    v[1]=max(x,intervals[i][1]);
                }else{
                    ans.push_back(intervals[i]);
                }
            }
        }
        
        return ans;
        
    }
};

File: merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        ans = []
        intervals.sort()

        for i in range(len(intervals)):
            if not ans:
                ans.append(intervals[i])
            else:
                x = ans[-1]

                if x[1] >= intervals[i][0]:
                    ans[-1][1] = max(x[1],intervals[i][1])
                else:
                    ans.append(intervals[i])
        
        return ans